File: src/test/java/io/plagov/advent/aoc2016/Day04Test.java
package io.plagov.advent.aoc2016;

import org.junit.jupiter.api.Test;
import util.FileUtil;

import java.util.List;

import static org.junit.jupiter.api.Assertions.assertEquals;

class Day04Test {

    private final List<String> input = FileUtil.readInputFile("2016/day04.txt");

    @Test
    void solveFirst() {
        var day04 = new Day04();
        assertEquals(173787, day04.partOne(input));
    }

    @Test
    void solveSecond() {
        var day04 = new Day04();
        assertEquals(548, day04.partTwo(input));
    }
}

File: src/test/kotlin/io/plagov/advent/aoc2015/Day04Test.kt
package io.plagov.advent.aoc2015

import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Test

class Day04Test {

  private val day4 = Day04()

  private val input = "iwrupvqb"

  @Test
  fun solveFirst() {
    assertEquals(346386, day4.partOne(input))
  }

  @Test
  fun solveSecond() {
    assertEquals(9958218, day4.partTwo(input))
  }
}
